### PR TITLE
ci: use pull_request_target for dependent action

### PR DESCRIPTION
**Pull request by leotrs**
## List of Changes
- Allow the dependent issues bot to sync PRs automatically
- Have the bot run every hour

## Motivation
Right now, there is no check for PRs. Thanks to @z0al for suggesting #802 .

cc @naveen521kk reivew plz?

### CODE REVIEW

**Comment by naveen521kk** [comment]
Why these many commits?

File: .github/workflows/dependent-issues.yml [hunk 1]
```diff
@@ -2,8 +2,18 @@ name: Dependent Issues
...
+      - reopened
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
   schedule:
-    - cron: '0/30 * * * *' # schedule daily check
+    - cron: '0/60 0 * * *' # schedule hourly check
```
Comment (naveen521kk) [inline .github/workflows/dependent-issues.yml:16]: I would suggest making it once a day. It's not required for running it every hour. Previously it didn't run on PR event so we have to do so. But here it is not required. Change it to once a day.

**Review by naveen521kk — CHANGES_REQUESTED** [review]

File: .github/workflows/dependent-issues.yml (hunk 1 above)
Comment (leotrs) [reply .github/workflows/dependent-issues.yml:16]: Ahh you're right. Can you suggest what is the right crontab for once daily?

**Comment by leotrs** [comment]
> Why these many commits?

Because I am bad at git :joy:

**Review by naveen521kk — APPROVED** [review]
LGTM!

**Pull request merged**